Done

johnwaynejr opened the pull request.

johnwaynejr committed:
Перенос готового проекта в новый из-за системных ошибок #1

johnwaynejr committed:
Перенос готового проекта в новый из-за системных ошибок #2

johnwaynejr committed:
Добавил файлы в Git

johnwaynejr committed:
Исправил TargetAPI с 31 на 32 в AndroidManifest.xml

johnwaynejr committed:
Добавил инициализацию RecycleView (треки начали подгружаться)

johnwaynejr committed:
Настраиваю холдер

johnwaynejr committed:
Подгрузил в холдер изображения

johnwaynejr committed:
Доработал обновление поиска

johnwaynejr committed:
Убрал кнопку, добавил реализацию с виртуальной клавиатуры

johnwaynejr committed:
Убрал кнопку, добавил реализацию с виртуальной клавиатуры

johnwaynejr committed:
Исправил отображение времени треков

johnwaynejr committed:
Исправил ошибки при отображени времени треков

johnwaynejr committed:
Создал кнопку "Обновить"

johnwaynejr committed:
Доработал кнопку "Обновить"

johnwaynejr committed:
Доработал отображение заглушек в темной теме

gusar-off commented on app/src/main/java/com/hfad/playlistmaker/FindActivity.kt:
```diff
@@ -0,0 +1,202 @@
+import android.view.ViewGroup
+import android.view.inputmethod.EditorInfo
+import android.view.inputmethod.InputMethodManager
+import android.widget.*
```
Wild card import-ы не советуют использовать в код стилях Kotlin. В настройках Android студии можно отключить

johnwaynejr replied:
Исправил

gusar-off commented on app/src/main/java/com/hfad/playlistmaker/iTunesApi.kt:
```diff
@@ -0,0 +1,10 @@
+import retrofit2.http.Query
+
+interface iTunesApi {
+    @GET("/search?entity")
```
⚠️По заданию get запросы надо слать c query параметром entity=song `/search?entity=song`, иначе апишка будет отдавать еще и другие типы контента.

johnwaynejr replied:
Исправил

johnwaynejr replied:
Исправил

gusar-off commented on app/src/main/java/com/hfad/playlistmaker/FindActivity.kt:
```diff
@@ -0,0 +1,202 @@
+            }
+
+            override fun onFailure(call: Call<SongResponse>, t: Throwable) {
+                trackList.visibility = View.GONE
```
Скрытие списка треков и показ заглушки лучше вынести в метод, чтобы избежать дублирования кода

johnwaynejr replied:
Исправил

johnwaynejr replied:
Исправил

gusar-off commented on app/src/main/java/com/hfad/playlistmaker/FindActivity.kt:
```diff
@@ -0,0 +1,202 @@
+        }
+    }
+
+    /*private fun showMessage(text: String, additionalMessage: String) {
```
Закомментированный код лучше убрать.

johnwaynejr replied:
Исправил

gusar-off reviewed (commented):
У тебя в пулл реквесте в изменениях также затянуты прошлые работы, нужно чтобы твой пр нес в себе только изменения выполненные в рамках задания, после принятия работы пр нужно вмержить в мастер, в результате в мастер ветке накапливаются все предыдущие задания, а в девелоп ветке ты разрабатываешь новые задания

johnwaynejr committed:
Исправил GET-запрос на  /search?entity=song
Настроил AS для работы без Wild card import-ов
Вынес скрытие списка треков и отображение кнопки-заглушки в отдельный метод
Убрал код комментарий

johnwaynejr committed:
Объединил методы скрытие списка треков и отображение кнопки-заглушки в отдельный метод
